runtime/lua/vim/lsp/buf.lua

```diff
@@ -100,7 +135,17 @@ function M.hover(config)
       return
     end
 
-    lsp.util.open_floating_preview(contents, format, config)
+    local _, winid = lsp.util.open_floating_preview(contents, format, config)
+
+    if config.apply_highlight then
+      api.nvim_create_autocmd('WinClosed', {
```

Comment:
Is this necessary? The buffer has `bufhidden = 'wipe'` set. Once the window closes the buffer disappears, and I'd think with it the highlights.

Reply:
I might be misunderstanding, but wouldn't this only apply to highlights within the hover buffer itself? This autocmd is for the regular buffer the user is on, i.e. whenever the hover window closes, remove the hover highlight from the other, regular code buffer